### ISSUE TITLE
machine_learning_day4

### ISSUE THREAD

**Issue by hallantern** [hallantern/hal_machine_learning#4]
# 深度学习
## 发展趋势
前沿
## 深度学习的三个步骤
Step1：神经网络（Neural network）
Step2：模型评估（Goodness of function）
Step3：选择最优函数（Pick best function）
### step1：神经网络
类比神经元
神经网络也可以有很多不同的连接方式，这样就会产生不同的结构（structure）在这个神经网络里面，我们有很多逻辑回归函数，其中每个逻辑回归都有自己的权重和自己的偏差，这些权重和偏差就是参数。 那这些神经元都是通过什么方式连接的呢？其实连接方式都是你手动去设计的。
[具体模型和深度的理解](https://datawhalechina.github.io/leeml-notes/#/chapter13/chapter13)
### step2：模型评估
对于模型的评估，我们一般采用损失函数来反应模型的好差，所以对于神经网络来说，我们采用交叉熵（cross entropy）函数来对y和y～的损失进行计算，接下来我们就是调整参数，让交叉熵越小越好。

对于损失，我们不单单要计算一笔数据的，而是要计算整体所有训练数据的损失，然后把所有的训练数据的损失都加起来，得到一个总体损失L。接下来就是在function set里面找到一组函数能最小化这个总体损失L，或者是找一组神经网络的参数θ，来最小化总体损失L。
### step3：选择最优函数
如何找到最优的函数和最好的一组参数呢，用的就是梯度下降（前一篇）
具体流程：θ是一组包含权重和偏差的参数集合，随机找一个初试值，接下来计算一下每个参数对应偏微分，得到的一个偏微分的集合∇L，就是梯度,有了这些偏微分，我们就可以不断更新梯度得到新的参数，这样不断反复进行，就能得到一组最好的参数使得损失函数的值最小

# 反向传播
反向传播（Backpropagation）是一个比较有效率的算法，让你计算梯度（Gradient） 的向量（Vector）时，可以有效率的计算出来

[具体看文档](https://datawhalechina.github.io/leeml-notes/#/chapter14/chapter14)，结合视频了解模型原理
